### ISSUE TITLE
[api] Section 11 PaymentResponse seems to be missing shippingAddress

### ISSUE THREAD

**Issue by halindrome** [w3c/browser-payment-api#75]
If the web site requests sets requestShipping to true, shouldn't the user's shipping address be returned as part of the PaymentResponse?

**Comment by rsolomakhin** [comment]
It's a part of the of the PaymentRequest object, I believe.

``` JavaScript
var request = new PaymentRequest(methods, details, {'requestShipping': true});
request.addEventListener('onShippingAddressChange', function(event) {
  console.log(request.shippingAddress);
});
request.show().then(handleResponse).catch(handleError);
```

**Comment by zkoch** [comment]
That's true, @rsolomakhin, but it still seems like it should be part of the PaymentResponse. PaymentResponse should contain the entirety of information you requested.

**Comment by halindrome** [comment]
I concur @zkoch.  It feels weird that we would be modifying the input object.  Actually I think the PaymentRequest object should NOT expose the shippingaddress.  That feels like a way less reuptable merchants could somehow mine customer address info even when a transaction did not complete (because event handlers get fired when address data is updated and thus could be harvested from the input object).

**Comment by dlongley** [comment]
The reason why the address is available in the event handler is because merchants may want to specify different shipping options based on the chosen address. But, yes, this could be used to mine customer addresses.

**Comment by dlongley** [comment]
I also wonder if users would be surprised to find out they had given their address info to the merchant site even if they didn't authorize payment.

**Comment by halindrome** [comment]
Yes.  And I get that.  It just seems like a weakness.  Note also that a merchant may want to offer different pricing based upon shipping address.  Wine.com, for example, charges different rates depending on the state you live in.

**Comment by dlongley** [comment]
@halindrome,

Yes, both are arguments for collecting shipping address earlier in the shopping process.

**Comment by rvm4** [comment]
Hmm, this is tricky. We cannot (in general) determine final pricing without knowing where the product is going which necessitates passing addresses to the client before authorization occurs, but we cannot get to the authorization screen without determining final pricing.

If I suppose that payment mediators are likely to implement this by presenting address forms prior to payment methods/apps, then I would argue that this is no less secure that standard practices today and consumers are used to the risk of providing an address. 

For privacy sake though, I would suggest we only pass in vague address information until payment has been authorized. For instance, giving the client country, state, and zip code is generally enough for any shipping purposes and does not leak very sensitive information (these are things you may already be able to deduce).

**Comment by adrianhopebailie** [comment]
@zkoch said:

> PaymentResponse should contain the entirety of information you requested.

I don't agree. It should be easy for the merchant to determine which data comes from the user agent (returned via the events) and which comes from the payment app (part of the payment response).

I think @rsolomakhin is correct here.

**Comment by adrianba** [comment]
Fixed by #139.
